flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java

```diff
@@ -1372,6 +1381,12 @@
 	/** The environment variable name which contains the location of the lib folder */
 	public static final String ENV_FLINK_LIB_DIR = "FLINK_LIB_DIR";
 
+	/** The environment variable name which contains the location of the bin directory */
+	public static final String ENV_FLINK_BIN_DIR = "FLINK_BIN_DIR";
+
+	/** The environment variable name which contains the Flink installation root directory */
+	public static final String ENV_FLINK_HOME_DIR = "FLINK_HOME";
+
```

Comment:
@Makman2  the standalone TM relies on the `FLINK_CONF_DIR` environment variable to find its configuration.  Now the Mesos TM does too, for consistency, rather than looking to the working directory.   

When the AM launches the TM as a Mesos task, it installs Flink to a subdirectory within the sandbox directory, and sets the `FLINK_HOME` environment variable accordingly.   The variable value is a relative path (based on the sandbox directory); it is assumed that the TM working directory is the sandbox directory.

Does this address your concern?